File: 剑指offer/变态跳台阶.java
public class Solution {
    public int JumpFloorII(int target) {
       if (target <= 0) return 0;
       if (target ==1) return 1;
        int a=1;
        int b=2;
        for(int i=2;i<=target;i++){
           b =2*a;
            a=b;
        }
        return b;
    }
}

File: 剑指offer/变态跳台阶.js
function jumpFloorII(number)
{
    // write code here
    if(number<=0) {
        return -1;
    }else if(number===1){
        return 1
    }else{
        return 2*jumpFloorII(number-1)
    }
    
}

function jumpFloorII(number)
{
    // write code here
    let i = 1;
    while(-- number) {
        i *= 2;
    }
    return i;
}